experiment/logviewer/gcsreader/parsing.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"bufio"
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/kubernetes/klog"
+)
+
+func processLines(reader io.Reader, regex *regexp.Regexp) ([]*logEntry, error) {
+	var result []*logEntry
+	r := bufio.NewReader(reader)
+	for {
+		line, err := r.ReadBytes('\n')
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
+		matched, entry, err := processLine(line, regex)
+		if err != nil {
+			// TODO There is a problem that files finish with incomplete line
+			klog.Errorf("%s error parsing line %s", err, line)
+		}
+		if matched && err == nil {
+			result = append(result, entry)
+		}
+	}
+	return result, nil
+}
+
+func processLine(line []byte, regex *regexp.Regexp) (bool, *logEntry, error) {
+	if !regex.Match(line) {
+		return false, nil, nil
+	}
+
+	parsed, err := parseLine(string(line))
+	return true, parsed, err
+}
+
+func parseLine(line string) (*logEntry, error) {
```

Comment:
In fact, audit logs are quite bad example of log format.
kube-apiserver.log is a better one.

We can leave it as is and change later.